src/app/components/shared/datetime-formats/datetime/datetime/datetime.component.spec.ts

```diff
@@ -0,0 +1,217 @@
+    expectedDateTimeAttribute: "2020-01-01T18:10:11.123+08:00",
+  },
+  {
+    name: "Luxon DateTime with dateOnly",
+    value: DateTime.fromISO("2020-01-01T12:10:11.123Z"),
+    expectedText: "2020-01-01",
+    expectedTooltip: "2020-01-01 20:10:11.123 +08:00 Australia/Perth",
+    expectedDateTimeAttribute: "2020-01-01T20:10:11.123+08:00",
+    date: true,
+  },
+  {
+    name: "Luxon DateTime with timeOnly",
+    value: DateTime.fromISO("2020-01-01T12:10:11.123Z"),
+    expectedText: "20:10:11",
+    expectedTooltip: "2020-01-01 20:10:11.123 +08:00 Australia/Perth",
+    expectedDateTimeAttribute: "2020-01-01T20:10:11.123+08:00",
+    time: true,
+  },
+  {
+    name: "Luxon DateTime with a negative offset (no timezone)",
+    value: DateTime.fromISO("2020-01-01T12:10:11.123-01:00"),
+    expectedText: "2020-01-01 21:10:11",
+    expectedTooltip: "2020-01-01 21:10:11.123 +08:00 Australia/Perth",
+    expectedDateTimeAttribute: "2020-01-01T21:10:11.123+08:00",
+  },
+  {
+    name: "Luxon DateTime with a western timezone",
+    value: DateTime.fromISO("2020-01-01T12:10:11.123Z").setZone(
+      "America/Los_Angeles",
+      { keepLocalTime: true }
+    ),
+    expectedText: "2020-01-02 04:10:11",
+    expectedTooltip: "2020-01-02 04:10:11.123 +08:00 Australia/Perth",
+    expectedDateTimeAttribute: "2020-01-02T04:10:11.123+08:00",
+  },
+  {
+    name: "JavaScript Date with no offset",
+    value: new Date("2020-01-01T12:10:11.123Z"),
+    expectedText: "2020-01-01 20:10:11",
+    expectedTooltip: "2020-01-01 20:10:11.123 +08:00 Australia/Perth",
+    expectedDateTimeAttribute: "2020-01-01T20:10:11.123+08:00",
+  },
+  {
+    name: "JavaScript Date with an offset",
+    value: new Date("2020-01-01T12:10:11.123+01:00"),
+    expectedText: "2020-01-01 19:10:11",
+    expectedTooltip: "2020-01-01 19:10:11.123 +08:00 Australia/Perth",
+    expectedDateTimeAttribute: "2020-01-01T19:10:11.123+08:00",
+  },
+  {
+    name: "JavaScript Date with a negative offset",
+    value: new Date("2020-01-01T12:10:11.123-01:00"),
+    expectedText: "2020-01-01 21:10:11",
+    expectedTooltip: "2020-01-01 21:10:11.123 +08:00 Australia/Perth",
+    expectedDateTimeAttribute: "2020-01-01T21:10:11.123+08:00",
+  },
+];
+
+describe("DatetimeComponent", () => {
+  let spectator: Spectator<DatetimeComponent>;
+  let mockSystemZone: string;
+
+  const createComponent = createComponentFactory({
+    component: DatetimeComponent,
+    imports: [SharedModule, MockBawApiModule],
+  });
+
+  function setup(): void {
+    // to ensure that tests pass for all timezones, we set the default timezone to Australia/Perth
+    // this mocks the test runners timezone to be Australia/Perth so that they will assert correctly in all timezones
+    Settings.defaultZone = "Australia/Perth";
+    mockSystemZone = "Australia/Perth";
+
+    spyOnProperty(SystemZone, "instance").and.callFake(() =>
+      IANAZone.create(mockSystemZone)
+    );
```

Comment:
why is this needed?

Reply:
furhter why is it in the general setup if it is only used in one test

Reply:
Since we're using `SystemZone.instance` to fetch the users local timezone in the `dateTime.component.ts`, we want to mock this getter so that it returns the mock timezone we are using for the `datetime.component.spec.ts` testbed.